### ISSUE TITLE
leccalc and the periods file

### ISSUE THREAD

**Issue by benhayes21** [OasisLMF/ktools#103]
leccalc produces some strange behavior when the periods.bin file is present. 

Using the make check files in the examples directory, if I add an example periods.bin file for the test model in the inputs directory (attached here [periods.tar.gz](https://github.com/OasisLMF/ktools/files/4230706/periods.tar.gz)) and run leccalc outputs, the headers are different if the file is present or not - specifically the "type" column is not there when the periods.bin file is present. 

The losses also look quite different, event though all period in the file have equal weighting - I'm not sure if this is expected behavior or not?

```
ben@ben:/tmp/ktools/ktools-3.2.2/examples$ head lec_fu_a_test.csv
summary_id,type,return_period,loss
1,1,10000.000000,9763.553711
1,1,7500.000000,8505.902344
1,1,5000.000000,7248.250488
1,1,1000.000000,5901.620117
1,1,500.000000,5039.050293
1,1,250.000000,4692.480469
1,1,200.000000,2335.742920
1,1,100.000000,0.000000
1,1,50.000000,0.000000
ben@ben:/tmp/ktools/ktools-3.2.2/examples$ head lec_fu_a_test_withperiod.csv 
summary_id,return_period,loss
1,10000.000000,28717.486328
1,7500.000000,27080.904297
1,5000.000000,24656.556641
1,1000.000000,13274.477539
2,10000.000000,34191.554688
2,7500.000000,32221.664062
2,5000.000000,29116.166016
2,1000.000000,14569.973633
2,500.000000,2658.760742
```

In addition, when I run the lec wheat-sheaf output option (-W), all losses are zero when the periods.bin file is present but there are losses when it is not. We also see the `-1` losses when the periods file is not there. I'm running 1,000 samples here, btw.

```
ben@ben:/tmp/ktools/ktools-3.2.2/examples$ head lec_ws_a_test.csv
summary_id,sidx,return_period,loss
1,1,10000.000000,32121.519531
1,1,7500.000000,27971.710938
1,1,5000.000000,23821.902344
1,1,1000.000000,9879.715820
1,1,500.000000,0.000000
1,1,250.000000,0.000000
1,1,200.000000,0.000000
1,1,100.000000,0.000000
1,1,50.000000,0.000000
ben@ben:/tmp/ktools/ktools-3.2.2/examples$ head -20 lec_ws_a_test_withperiod.csv
summary_id,sidx,return_period,loss
1,-1,10000.000000,0.000000
1,-1,7500.000000,0.000000
1,-1,5000.000000,0.000000
1,-1,1000.000000,0.000000
1,-1,500.000000,0.000000
1,-1,250.000000,0.000000
1,-1,200.000000,0.000000
1,-1,100.000000,0.000000
1,-1,50.000000,0.000000
1,-1,25.000000,0.000000
1,-1,20.000000,0.000000
1,-1,10.000000,0.000000
1,-1,5.000000,0.000000
1,-1,2.000000,0.000000
1,1,10000.000000,0.000000
1,1,7500.000000,0.000000
1,1,5000.000000,0.000000
1,1,1000.000000,0.000000
1,1,500.000000,0.000000
ben@ben:/tmp/ktools/ktools-3.2.2/examples$ cut -f4 -d, lec_ws_a_test_withperiod.csv | sort -u
0.000000
loss
```

**Comment by johcarter** [comment]
I've tested the examples in ktest with your neutral periods file (all weightings = 1/P). 

The expected result is that all leccalc outputs will be the same as if there were no periods file with only floating point differences in losses.

Here are the issues i am seeing:

**full uncertainty**
1) when using a periods file the fields output are summary_id, return_period, loss (wrong) and without a periods file they are summary_id, type, return_period, loss (right). 
The periods branch of the code was created before we updated the output format to split out sidx -1 (type 1) from the other samples (type2) and has not been updated. therefore the curve needs to be decoupled into type 1 and type 2 on the periods branch.

**wheatsheaf**
1) when using a periods file we are getting results for sidx -1 (wrong) and without a periods file we are not getting results for sidx -1 (right).
The periods branch of the code was created before we updated the wheatsheaf to remove sidx -1  so we need to update the periods branch to remove sidx -1.
2) when using the return period switch -r with a periods file present, the losses look wrong. not sure why yet, but suggest fixing 1) first and see if that solves the problem

**wheatsheaf mean**
looks ok with and without periods file, only floating point differences in losses

**sample mean**
looks ok with and without periods file, only floating point differences in losses
